### DISCUSSION TITLE
Never refresh after the component be unmouned

### DISCUSSION THREAD

**Discussion by ethever** [pmndrs/jotai#2060] [Bug report]
## Summary
![image](https://github.com/pmndrs/jotai/assets/25705974/5bf0266e-6bf4-4588-86c0-e84a4c5bb0e4)
First, click `mutate` button.
Then immediately click `toggle` button:
![image](https://github.com/pmndrs/jotai/assets/25705974/33e6491e-9d0f-4595-a00d-ed5ad6f26395)

finally the `state` of `TestAtom` will be lock in `loading` forever.

``` typescript
 const [state, mutate] = useAtom(TestAtom);
```

## Link to reproduction
https://github.com/ethever/jotai-react-bug

## Check List

Please do not ask questions in issues.

- [x ] I've already opened a [discussion](https://github.com/pmndrs/jotai/discussions) before opening this issue, or already discussed in other media.

Please include a minimal reproduction.

- [x ] I've added a link to a [typescript playground](https://www.typescriptlang.org/play) or [codesandbox](https://codesandbox.io) with a minimal reproduction.

**Comment by dai-shi** [comment]
Thanks for reporting!
Would anyone like to dig into it?

My guess is this is a bug or a limitation in jotai-tanstack-query, which is kind of not too surprising as we want to redesign it. https://github.com/jotaijs/jotai-tanstack-query/issues/32#issuecomment-1523619697

If you can reproduce the the issue without jotai-tanstack-query, there might be a bug in jotai core, and it's more serious.

**Comment by dai-shi** [comment]
https://github.com/jotaijs/jotai-tanstack-query/pull/38 by @Aslemammad
